Paylike/Transaction.php

```diff
@@ -0,0 +1,109 @@
+<?php
+namespace Paylike;
+/**
+ * Class Transaction
+ * @package Paylike
+ * Handles transaction operations.
+ *
+ * @version    1.0.0
+ */
+class Transaction {
+
+	/**
+	 * @param $transaction_id
+	 *
+	 * @return int|mixed
+	 * Return the transaction data
+	 *
+	 */
+	public static function fetch( $transaction_id ) {
+		$adapter = Client::getAdapter();
+		if ( ! $adapter ) {
+			return - 1;
+		}
+
+		return $adapter->request( 'transactions/' . $transaction_id );
+
+	}
+
+
+	/**
+	 * @param $transaction_id
+	 * The authorization takes place via the client script,
+	 * so this actually returns the transaction data
+	 *
+	 * @return int|mixed
+	 */
+	public static function authorize( $transaction_id ) {
+		return self::fetch( $transaction_id );
+	}
+
+
+	/**
+	 * @param $transaction_id
+	 * @param $amount
+	 * @param $currency
+	 * @param string $description
+	 * Capture a transaction that has been authorized.
+	 * This also returns the transaction data.
+	 *
+	 * @return bool|int|mixed
+	 */
+	public static function capture( $transaction_id, $amount, $currency, $description = '' ) {
```

Comment:
"description" and "descriptor" is *not* the same. "descriptor" is the text on the customers bank statement, and it has to comply with [rigged rules](https://github.com/paylike/descriptor), failing to comply will result in an error response from the API.
It is better not to provide this than providing something which has not been validated and carefully constructed according to the guidelines. There is a fallback on each merchant which most often is the name of the shop or similar.